Makefile

```diff
@@ -604,6 +605,26 @@ $(SBOM): $(ROOTFS_TAR) | $(INSTALLER)
 	rm -rf $(TMP_ROOTDIR)
 	$(QUIET): $@: Succeeded
 
+$(GENERATE_METADATA): $(ROOTFS_TAR) | $(INSTALLER)
+	$(QUIET): $@: Begin
+	$(eval TMP_ROOTDIR := $(shell mktemp -d))
+	# this is a bit of a hack, but we need to extract the rootfs tar to a directory, and it fails if
+	# we try to extract character devices, block devices or pipes, so we just exclude the dir.
+	# when syft supports reading straight from a tar archive with duplicate entries,
+	# this all can go away, and we can read the rootfs.tar
+	# see https://github.com/anchore/syft/issues/1400
+	tar xf $< -C $(TMP_ROOTDIR) --exclude "dev/*"
+	bash tools/get-alpine-pkg-source.sh -v -s /tmp/sources/alpine -e $(TMP_ROOTDIR) 
+	bash tools/get-kernel-source.sh -v -s /tmp/sources/kernel 
```

Comment:
Is it useful to pass -v to have the build spew out so many lines of output?

Reply:
fixed